gapic/src/main/com/google/gapid/perfetto/models/FrameInfo.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.perfetto.models;
+
+import static com.google.gapid.util.MoreFutures.transform;
+
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gapid.models.Perfetto;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Data about a GPU in the trace.
+ */
+public class FrameInfo {
+  public static final FrameInfo NONE = new FrameInfo(Collections.emptyList());
+
+  private static final String FRAME_SLICE_QUERY =
+      "select t.id, t.name, t.scope, max(depth) + 1 " +
+      "from gpu_track t join frame_slice s on (t.id = s.track_id) " +
```

Comment:
This doesn't look right. Shouldn't this be limited to a different table than gpu_track?

Reply:
Good catch actually. I think moving this out of gpu_track is going to involve a bit of changes in perfetto judging by the places its being touched. I'll create a follow up bug and once that is done, I'll patch this separately? This change has some important UI changes that I want to get in first.